dataprep/clean/clean_url.py

```diff
@@ -0,0 +1,197 @@
+# pylint: disable=too-many-arguments, trailing-whitespace, global-statement
+"""
+implement clean_url functionality
+"""
+
+import re
+from typing import Any, Union, Dict, List
+from urllib.parse import unquote, urlparse
+
+import pandas as pd
+import dask
+import dask.dataframe as dd
+
+from .utils import NULL_VALUES, to_dask
+
+# to extract queries
+QUERY_REGEX = r"(\?|\&)([^=]+)\=([^&]+)"
+
+# regex to validate url
+VALID_URL_REGEX = r"^(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&'\(\)\*\+,;=.]+$"
+
+# removing auth params
+AUTH_VALUES = {"access_token", "auth_key", "auth", "password", "username", "login", "token"}
+
+# unified_list
+UNIFIED_AUTH_LIST = set()
+
+# STATs count
+STATS = {"cleaned": 0, "rows": 0}
+
+
+def clean_url(
+    df: Union[pd.DataFrame, dd.DataFrame],
+    column: str,
+    inplace: bool = False,
+    split: bool = False,
+    remove_auth: Union[bool, List[str]] = False,
+    report: bool = True,
+) -> Union[pd.DataFrame, dd.DataFrame]:
+
+    """
+    This function cleans url
+    Parameters
+    ----------
+    df
+        pandas or Dask DataFrame
+    column
+        column name
+    split
+        If True, split a column containing into the scheme, hostname, queries, cleaned_url columns
+        if set to False would return a new column of dictionaries with the relavant
+        information (scheme, host, etc.) in form of key-value pairs
+    inplace
+        If True, delete the given column with dirty data, else, create a new
+        column with cleaned data.
+    remove_auth
+        can be a bool, or list of string representing the names of Auth queries
+        to be removed. By default it is set to False
+    report:
+        Displays how many queries were removed from rows
+    """
+
+    # reset stats and convert to dask
+    reset_stats()
+    df = to_dask(df)
+
+    # unified list of auth removal params
+    if not isinstance(remove_auth, bool):
+        global UNIFIED_AUTH_LIST
+        UNIFIED_AUTH_LIST = {*AUTH_VALUES, *set(remove_auth)}
+
+    # specify the metadata for dask apply
+    meta = df.dtypes.to_dict()
+
+    if split:
+        meta.update(zip(("scheme", "host", "cleaned_url", "queries"), (str, str, str, str, str)))
+    else:
+        meta.update(zip(("url_details",), (str,)))
+
+    df = df.apply(
+        format_url,
+        args=(column, split, remove_auth),
+        axis=1,
+        meta=meta,
+    )
+
+    df = dask.compute(df, df.shape[0])[0]
+
+    if inplace:
+        if split:
+            df.drop(columns=[column])
+        else:
+            df.drop(columns=[column, "url_details"])
```

Comment:
When calling `df.drop`, it will return new df while not modifying initial df, so `inplace` parameter will not work here. Btw, do we intend to drop both `columns=[column, "url_details"]` here? When `inplace == True and split == False`, it will give us the empty Dataframe I think. Can we create a new column for cleaned url and change the way to deal with `inplace` parameter?

Reply:
I have corrected this as well. The previous version was not returning the updated dataframe.